### DISCUSSION TITLE
Can't Handle a Request That Contains Non-ASCII Character (Emoji)

### DISCUSSION THREAD

**Discussion by rafazzaky** [micro/go-micro#2731] [General]
So I'm having this problem where my api-gateway (that use go-micro) can't handle a request that contain an emoji. The error message that I receive is `\"id\":\"go.micro.client\",\"code\":500,\"detail\":\"header key \\\"request\\\" contains value with non-printable ASCII characters\"` I was wondering is this go-micro related or something else? I hope I can find an answer in this discussion. My current solution is to encode the request and then decode it in my api-gateway.

**Comment by asim** [comment]
I've never seen an emoji in a http header. Is that the case?

**Reply by rafazzaky** [reply]
No, the case is that I use an emoji on the request body but somehow it identified as a header, for example, when I send this request
```
{
    "message":"this is a message"
}
```
it will work, but somehow when I send this request
```
{
    "message":"this is a message with emoji🙏"
}
```
it wont work and responded with an error `"contains value with non-printable ASCII characters"`
